### PR TITLE
get rid NodeSandbox.overrideElement function (close #464)

### LINKED ISSUE #464: Optimization - try to get rid of 'overrideElement' methods using prototypes

### PR DESCRIPTION
**Pull request by miherlosev**
1)method overriding was moved to prototype level
2)rename `NodeSandbox.overrideElement` to `NodeSandbox.processDOM` (because now it contains only adding services handlers for img, iframe, file inputs and write element's processing context)
3)small IframeSandbox refactoring

### CODE REVIEW

**Comment by inikulin** [comment]
@miherlosev 
> rename NodeSandbox.overrideElement to NodeSandbox.processDOM (because now it contains only adding services handlers for img, iframe, file inputs and write element's processing context)

Maybe `NodeSandbox.setup` or smthg like that?

**Comment by miherlosev** [comment]
So then we should rename the following entities:
`INTERNAL_PROPS.overrideDomMethodName` -> `INTERNAL_PROPS.setupDomMethodName`
`EventSandbox.processElement` -> `EventSandbox.setupElement`
`IframeSandbox.processIframe` -> `IframeSandbox.setupIframe`
`ElementSandbox.processElement` -> `ElementSandbox.setupElement`.

because they are nested sandboxes for `NodeSandbox`.

**Comment by inikulin** [comment]
nope, it's just weird that `NodeSandbox` processes DOM. Name it `processNodes` then or smthg like that

**Comment by miherlosev** [comment]
FPR

**Comment by miherlosev** [comment]
ping @churkin @inikulin

**Comment by inikulin** [comment]
lgtm

File: src/client/sandbox/event/index.js [hunk 1]
```diff
@@ -152,31 +150,27 @@ export default class EventSandbox extends SandboxBase {
...
-        if (ELEMENT_HAS_ADDITIONAL_EVENT_METHODS) {
-            this._overrideElementOrHTMLElementMethod('fireEvent', this.overridedMethods.fireEvent);
-            this._overrideElementOrHTMLElementMethod('attachEvent', this.overridedMethods.attachEvent);
-            this._overrideElementOrHTMLElementMethod('detachEvent', this.overridedMethods.detachEvent);
+        window.Document.prototype.dispatchEvent                = this.overridedMethods.dispatchEvent;
+        window.HTMLElement.prototype.dispatchEvent             = this.overridedMethods.dispatchEvent;
+        window.HTMLElement.prototype.focus                     = this.overridedMethods.focus;
+        window.HTMLElement.prototype.blur                      = this.overridedMethods.blur;
+        window.HTMLElement.prototype.click                     = this.overridedMethods.click;
+        window.focus                                           = this.overridedMethods.focus;
```
Comment (churkin) [inline src/client/sandbox/event/index.js]: Why not use prototype?

File: src/client/sandbox/iframe.js [hunk 2]
```diff
@@ -132,24 +132,14 @@ export default class IframeSandbox extends SandboxBase {
     }
 
     iframeAddedToDom (el) {
```
Comment (churkin) [inline src/client/sandbox/iframe.js]: excess method

File: src/client/sandbox/iframe.js [hunk 3]
```diff
@@ -132,24 +132,14 @@ export default class IframeSandbox extends SandboxBase {
...
-        this.overrideIframe(el);
-
-        if (!isShadowUIElement(el)) {
-            this._raiseReadyToInitEvent(el);
```
Comment (churkin) [inline src/client/sandbox/iframe.js]: Leave this code, move in the `processIframe`

**Comment by churkin** [comment]
-\r

**Comment by churkin** [comment]
I think this will give a very good performance boost!

**Comment by miherlosev** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Pull request merged**